### DISCUSSION TITLE
`get*` vs `*`

### DISCUSSION THREAD

**Discussion by ipavlic** [js-fns/date-fns#2645] [Ideas] [answered]
What is the reasoning for get prefixes of functions that operate on dates? For example, `getUnixTime` vs just `unixTime`. This seems inconsistent with, for example `difference*` functions which operate on two arguments, or `endOf*` functions, which also operate on a single date argument. `get` prefixes appear more appropriate for OOP methods, as opposed to just functions.

**Comment by tan75** [comment, marked as answer]
Hi @ipavlic 
thank you for such an interesting question! 
I have already responded to this in the issue you raised so I'll just do a copy-paste here.

There are few reasons for having our current naming convention as is.
We wanted to make our API as similar to the standard JS library as possible so both APIs could easily be used together.
When it comes to `getUnixTime`,  it copies the convention from [MDN JS Reference Date getTime](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime)
One of the reasons for this naming convention is that its easier to maintain this huge amount of functions that we currently have: e.g. 
its easier to implement mass changes (e.g. typescriptify) if all the functions of the same type have the same prefix hence 
located together in Finder.
All  the `get*` functions accept one argument and return an integer hence for example, we've got `differenceInDays`, not ~getDifferenceInDays~.
Hope this clarifies your query.

**Reply by ipavlic** [reply]
Hi! Thank you for your response. I appreciate all the hard work put into this library. My view on the difference between `Date.getTime` and, for example, `getUnixTime` is that `getTime` is a member function. So one would typically have something like `const d = new Date()`, and then call `d.getTime()`. `getUnixTime` on the other hand doesn't get called on an instance — it's just a regular function. That's commonly the difference between what I labeled as an "object oriented programming" convention and "functional programming" conventions. 

The chosen convention does not seem to be consistent either. For example, `getOverlappingDaysInIntervals` takes two `Interval`s and returns an integer. `differenceInDays` takes two `Date`s and returns an integer. They do similar things, yet one has a `get` prefix, while the other does not.

**Reply by tan75** [reply]
@ipavlic 
We are doing our best to make this library consistent and easy to use.
But the issue is that currently we have a huge amount of functions and it's not quite easy to make the API consistent within all the functions currently available. 
Take the below functions for example: 
The `getMilliseconds` function returns milliseconds of a given date and the `milliseconds` function returns milliseconds of a given duration.
Here the `get` prefix allows us to differentiate those 2 functions. 
Thank you for your opinion and interest in making this library better - we will definitely consider your suggestions.

**Reply by ipavlic** [reply]
A potential solution for future functions might something like `dateMilliseconds` and `durationMilliseconds`. On the other hand I understand that this would get very cumbersome over time. Thanks again!
